### PR TITLE
Adding pkeyutl tool to the CLI

### PR DESCRIPTION
**Pull request by smittals2**
### Description of changes: 
Add pkeutl tool with: 
-help
-in
-out
-sign
-verify
-inkey
-sigfile
-pubin
-passin 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tool-openssl/pkeyutl.cc [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    if (!WriteOutput(signature, out_path, hexdump)) {
+      return false;
+    }
+  } else if (verify) {
+    // For verification, we need a signature file or signature data
+    // For simplicity, we'll read signature from stdin if no specific sig file
+    std::vector<uint8_t> signature;
+    if (!ReadInputData("", signature)) { // Read signature from stdin
+      return false;
+    }
```
Comment (kingstjo) [inline tool-openssl/pkeyutl.cc]: OpenSSL 1.1.1 requires a -sigfile parameter for verification operations, but this implementation reads signatures from stdin instead.

Would you consider adding -sigfile support for compatibility?

**Comment by justsmth** [comment]
The msys2/ucrt64 CI failures seem related:
```
tool-openssl/tool_openssl_test failed to complete: exit status 0xc0000005
```

A "0xc0000005" exit status seems to be associated with memory access violations on Windows.

**Comment by smittals2** [comment]
> The msys2/ucrt64 CI failures seem related:
> 
> ```
> tool-openssl/tool_openssl_test failed to complete: exit status 0xc0000005
> ```
> 
> A "0xc0000005" exit status seems to be associated with memory access violations on Windows.

Yea I've been trying to debug why we see this. Before opening the file, we stat it to make sure it exists. In this particular use case, the file is created by the test harness, opened via BIO APIs and populated by the pkeyutl tool, and then fails when we try to read the file subsequently to validate it's contents. I added additional explicit instructions to close any BIO handles that may create resource contention but continued seeing this error. 

We can see this in the logs here: 
```
> DEBUG: ReadFileToString - Starting for path: C:\Users\RUNNER~1\AppData\Local\Temp\aws91A.tmp
> DEBUG: ReadFileToString - Checking if file exists with stat()
> DEBUG: ReadFileToString - File exists, size: 32 bytes
> DEBUG: ReadFileToString - Opening file stream
> tool-openssl/tool_openssl_test
> tool-openssl/tool_openssl_test failed to complete: exit status 0xc0000005
```

We also only see this access violation in msys2 ucrt64 - (Ninja/MinGW) builds. For example msys2 ucrt64 with MSYS Makefiles doesn't fail (neither do any of the other Windows CI checks). 

My two pending investigation avenues are as follows: 
1) Looking at the Cmake, the two build dimensions we see the error in are special cased: https://github.com/aws/aws-lc/blob/1a5b9eb7e97b8e0ea75d5e045bd7fbb01cbb38e1/.github/workflows/windows-alt.yml#L244. Maybe the diff shell is impacting which libraries our test executable links to
2) Using the BIO APIs to access the same tmp files doesn't cause an issue, but using ReadFileToString which uses the native C++ file reading APIs does, so maybe there's some edge case/bug in these two build dimensions.

**Review by justsmth — APPROVED** [review]

**Review by justsmth — DISMISSED** [review]

File: tool-openssl/pkeyutl_test.cc [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  int openssl_result = system(openssl_command.c_str());
+  ASSERT_EQ(openssl_result, 0) << "OpenSSL command failed: " << openssl_command;
+
+  // Verify both signatures with the public key
+  std::string tool_verify_cmd = std::string(tool_executable_path) + " pkeyutl -verify -pubin -inkey " + 
+                                pubkey_path + " -in " + in_path + " -sigfile " + sig_path_tool + 
+                                " > " + out_path_tool;
+  std::string openssl_verify_cmd = std::string(openssl_executable_path) + " pkeyutl -verify -pubin -inkey " + 
+                                   pubkey_path + " -in " + in_path + " -sigfile " + sig_path_openssl + 
+                                   " > " + out_path_openssl;
```
Comment (kingstjo) [inline tool-openssl/pkeyutl_test.cc:258]: Could we verify AWS-LC signature with OpenSSL `pkeyutl -verify` and vice-versa?
// 1. AWS-LC signs → OpenSSL verifies
// 2. OpenSSL signs → AWS-LC verifies
